src/upgrader.ts

```diff
@@ -258,39 +258,51 @@ export class DefaultUpgrader extends EventEmitter<UpgraderEvents> implements Upg
 
     log('Starting the outbound connection upgrade')
 
+    // If the transport natively supports encryption, skip connection
+    // protector and encryption
+
     // Protect
     let protectedConn = maConn
-    const protector = this.components.getConnectionProtector()
+    if ((opts?.muxerFactory) == null) {
+      const protector = this.components.getConnectionProtector()
 
-    if (protector != null) {
-      protectedConn = await protector.protect(maConn)
+      if (protector != null) {
+        protectedConn = await protector.protect(maConn)
+      }
     }
 
     try {
       // Encrypt the connection
-      ({
-        conn: encryptedConn,
-        remotePeer,
-        protocol: cryptoProtocol
-      } = await this._encryptOutbound(protectedConn, remotePeerId))
+      encryptedConn = protectedConn
+      if (!opts?.skipEncryption) {
```

Comment:
I think this is a bit easier to read.  The linter will also probably complain about not handling boolean values explicitly.

```suggestion
      if (opts?.skipEncryption === false) {
```

Reply:
Wouldn't this return true if `opts` is undefined or `opts.skipEncryption` is undefined?

Reply:
What I mean is that we want the default behavior to be to not skip encryption so we should do `opts?.skipEncryption !== true`. [example](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABFATjCBrAngCjgBygGcB+ALkQG8iMZ9yAjOOAGwFMBDMAXwEoqAUImEiYwRHkKkAdDTqIAvEsTAOLIm36UhI3RARFWbaSzgBzHACIAcnCiI5+fDDBnLvHbu6I26toN09AyMTcytHZ1d3TxFuATiBUEhYBEQGNigoNhRJYnIqR0Zmdi4+AMCxCQI82Vp8RABCZVQQTXLAxH0wQ3ZQixs7BzrItw8O718NdsCunuNTfoiXUZjhOITUdGwcD03MXEo+AQF0zOydk4ysnMPeIA)